felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1013,7 +1013,9 @@ func (m *bpfEndpointManager) calculateTCAttachPoint(policyDirection PolDirection
 	ap.PSNATEnd = m.psnatPorts.MaxPort
 	ap.IPv6Enabled = m.ipv6Enabled
 	ap.MapSizes = m.bpfMapContext.MapSizes
-	if m.Features != nil {
+	if m.Features == nil {
```

Comment:
Imo beased on this, you can safely assume that features are available https://github.com/projectcalico/calico/blob/master/felix/dataplane/linux/int_dataplane.go#L396-L397

Reply:
without checking for `nil`, one of the ut tests fails with a panic about accessing this address.

Reply:
github.com/projectcalico/calico/felix/dataplane/linux.(*bpfEndpointManager).calculateTCAttachPoint(0xc002081b80, 0x1, {0x1eb60cf, 0x0})
    /go/src/github.com/projectcalico/calico/felix/dataplane/linux/bpf_ep_mgr.go:1021 +0x3e4
github.com/projectcalico/calico/felix/dataplane/linux.(*bpfEndpointManager).attachDataIfaceProgram(0xc002081b80, {0x1eb60cf, 0x4}, 0x0, 0x0)
    /go/src/github.com/projectcalico/calico/felix/dataplane/linux/bpf_ep_mgr.go:894 +0x92
github.com/projectcalico/calico/felix/dataplane/linux.(*bpfEndpointManager).applyProgramsToDirtyDataInterfaces.func1.1()
    /go/src/github.com/projectcalico/calico/felix/dataplane/linux/bpf_ep_mgr.go:590 +0x4d3
created by github.com/projectcalico/calico/felix/dataplane/linux.(*bpfEndpointManager).applyProgramsToDirtyDataInterfaces.func1
    /go/src/github.com/projectcalico/calico/felix/dataplane/linux/bpf_ep_mgr.go:561 +0x19d

Reply:
that's why I felt it would safer to check for it. WDYT?

Reply:
right, fixed it. Needed to update a ut test as well.

Reply:
yeah, better to fix a test :-) Thanks!